rules/slack_rules/slack_user_privilege_escalation.py

```diff
@@ -13,16 +13,34 @@ def rule(event):
 
 
 def title(event):
-    if event.get("action") in USER_PRIV_ESC_ACTIONS:
-        return USER_PRIV_ESC_ACTIONS.get(event.get("action"))
+    username = deep_get(event, "actor", "user", "name", default="<unknown-actor>")
+    email = deep_get(event, "actor", "user", "email", default="<unknown-email>")
+
+    if event.get("action") == "owner_transferred":
+        return f"Slack Owner Transferred from {username} ({email})"
+
+    if event.get("action") == "permissions_assigned":
+        return f"Slack User, {username} ({email}), assigned permissions"
+
+    if event.get("action") == "role_change_to_admin":
+        return f"{username} ({email}) promoted to admin"
+
+    if event.get("action") == "role_change_to_owner":
+        return f"{username} ({email}) promoted to Owner"
+
     return "Slack User Privilege Escalation"
 
 
 def severity(event):
     # Downgrade severity for users assigned permissions
-    # TODO: Add case to check for admin privileges to escalate to Critical
     if event.get("action") == "permissions_assigned":
         return "Medium"
+    if event.get("action") == "role_change_to_admin":
+        return "Critical"
+    if event.get("action") == "role_change_to_owner":
+        return "Critical"
+    if event.get("action") == "owner_transferred":
+        return "Critical"
     return "High"
```

Comment:
```suggestion
    return "Critical"
```

Reply:
There are four events in the list, and there is no way to reach line 44 for High severity

Reply:
Ahh this makes a good point. Everything else is Critical except for "permission_assigned". Thanks for pointing this out!